Support new Linux kernel timers

Compiling under `Linux raspberrypi 4.19.58-v7+` produces the following:

```
make -C /lib/modules/4.19.58-v7+/build M=/home/pi/piProx-oss modules
make[1]: Entering directory '/usr/src/linux-headers-4.19.58-v7+'
  CC [M]  /home/pi/piProx-oss/piProx.o
/home/pi/piProx-oss/piProx.c: In function 'piProx_timer_init':
/home/pi/piProx-oss/piProx.c:262:2: error: implicit declaration of function 'setup_timer'; did you mean 'setup_irq'? [-Werror=implicit-function-declaration]
  setup_timer( &timer, piProx_timer_callback, 0 );
  ^~~~~~~~~~~
  setup_irq
cc1: some warnings being treated as errors
make[2]: *** [scripts/Makefile.build:310: /home/pi/piProx-oss/piProx.o] Error 1
make[1]: *** [Makefile:1517: _module_/home/pi/piProx-oss] Error 2
make[1]: Leaving directory '/usr/src/linux-headers-4.19.58-v7+'
make: *** [Makefile:7: modules] Error 2
```

I believe the interface for Linux kernel timers [has changed](https://lwn.net/Articles/735887/).  Looks like the new function is [timer_setup](https://github.com/torvalds/linux/blob/609488bc979f99f805f34e9a32c1e3b71179d10b/include/linux/timer.h#L140):

```c
/**
 * timer_setup - prepare a timer for first use
 * @timer: the timer in question
 * @callback: the function to call when timer expires
 * @flags: any TIMER_* flags
 *
 * Regular timer initialization should use either DEFINE_TIMER() above,
 * or timer_setup(). For timers on the stack, timer_setup_on_stack() must
 * be used and must be balanced with a call to destroy_timer_on_stack().
 */
#define timer_setup(timer, callback, flags)			\
	__init_timer((timer), (callback), (flags))
```

--- a/Makefile
+++ b/Makefile
@@ -1,11 +1,12 @@
 obj-m += piProx.o
+UNAME := $(shell uname -r)
 
 all: modules printhex
  
 modules: piProx.c
-	$(MAKE) -C /lib/modules/$(shell uname -r)/build M=$(shell pwd) modules 
+	$(MAKE) -C /lib/modules/$(UNAME)/build M=$(CURDIR) modules 
 clean:
-	$(MAKE) -C /lib/modules/$(shell uname -r)/build M=$(shell pwd) clean
+	$(MAKE) -C /lib/modules/$(UNAME)/build M=$(CURDIR) clean
 	$(RM) printhex
 
 printhex: libpiprox.c printhex.c

--- a/Makefile
+++ b/Makefile
@@ -3,9 +3,9 @@ obj-m += piProx.o
 all: modules printhex
  
 modules: piProx.c
-	$(MAKE) -C /lib/modules/$(shell uname -r)/build M=$(PWD) modules 
+	$(MAKE) -C /lib/modules/$(shell uname -r)/build M=$(shell pwd) modules 
 clean:
-	$(MAKE) -C /lib/modules/$(shell uname -r)/build M=$(PWD) clean
+	$(MAKE) -C /lib/modules/$(shell uname -r)/build M=$(shell pwd) clean
 	$(RM) printhex
 
 printhex: libpiprox.c printhex.c

--- a/piProx.c
+++ b/piProx.c
@@ -23,7 +23,7 @@
 #include <linux/timer.h>
 #include <linux/string.h>
 #include <linux/fs.h>
-#include <asm/uaccess.h>
+#include <linux/uaccess.h>
 #include <linux/device.h>
 #include <linux/wait.h>
 

--- a/piProx.c
+++ b/piProx.c
@@ -190,7 +190,7 @@ static void bit_set(char *buf, short int loc){
 	buf[byte] |= (1<<bit);	
 }
 // ISR Functions
-void piProx_timer_callback( unsigned long data )
+void piProx_timer_callback( struct timer_list *event )
 {
 	size_t bytes;
 	int i;
@@ -259,7 +259,7 @@ static irqreturn_t low_ISR(int irq, void *dev_id, struct pt_regs *regs) {
 
 //Initializes timer
 void piProx_timer_init(void){
-	setup_timer( &timer, piProx_timer_callback, 0 );
+	timer_setup( &timer, piProx_timer_callback, 0 );
 	printk(KERN_NOTICE "piProx: Starting timer\n");
 }
 

